src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random;
+    private ColorSupplier colorSupplier;
+
+    FigureSupplier() {
+        random = new Random();
+        colorSupplier = new ColorSupplier();
+    }
+
+    public Figure getRandomFigure() {
+        Figure[] list = new Figure[5];
+        list[0] = new Circle(colorSupplier.getRandomColor(), random.nextDouble());
+        list[1] = new IsoscelesTrapezoid(colorSupplier.getRandomColor(), random.nextDouble(),
+                random.nextDouble(), random.nextDouble());
+        list[2] = new RightTriangle(colorSupplier.getRandomColor(), random.nextDouble(),
+                random.nextDouble());
+        list[3] = new Square(colorSupplier.getRandomColor(), random.nextDouble());
+        list[4] = new Rectangle(colorSupplier.getRandomColor(), random.nextDouble(),
+                random.nextDouble());
+        return list[random.nextInt(list.length)];
```

Comment:
Don't create extra useless objects and array, use `switch-case` statement instead.